### PR TITLE
feat: install apiserver-network-proxy when create virtualCluster

### LINKED ISSUE #568: feature: hostPortManager support multi port for one virtualCluster
**What would you like to be added**:
hostPortManager can allocate multi port for same virtualCluster

**Why is this needed**:
api-server-networkproxy need extras port to connect between server and agent

### PR DESCRIPTION
**Pull request by village-way**
#### What type of PR is this?

/kind feature

#### Which issue(s) does this PR fix?

Fixes #568 #571 
#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/kubenest/manifest/controlplane/apiserver/mainfests_deployment.go [hunk 1]
```diff
@@ -128,5 +128,356 @@ spec:
...
+        secret:
+          defaultMode: 420
+          secretName: {{ .KubeconfigSecret }}
+      - name: varlogkonnectivityserver
+        hostPath:
+          path: /var/log/konnectivity-server.log
+          type: FileOrCreate
+      - name: kubernetes
+        hostPath:
+          path: /etc/kubernetes
```
Comment (duanmengkk) [inline pkg/kubenest/manifest/controlplane/apiserver/mainfests_deployment.go]: Is this necessary？

File: pkg/kubenest/manifest/controlplane/apiserver/mainfests_deployment.go [hunk 2]
```diff
@@ -128,5 +128,356 @@ spec:
...
+          type: DirectoryOrCreate
+      - name: apiserver-cert
+        secret:
+          secretName: {{ .VirtualClusterCertsSecret }}
+      - name: etcd-cert
+        secret:
+          secretName: {{ .EtcdCertsSecret }}
+      - name: konnectivity-uds
+        hostPath:
+          path: /etc/kubernetes/konnectivity-server/{{ .Namespace }}
```
Comment (duanmengkk) [inline pkg/kubenest/manifest/controlplane/apiserver/mainfests_deployment.go]: Namespace +  Name ？

File: pkg/kubenest/tasks/anp.go [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+	labelSelector := metav1.FormatLabelSelector(deployment.Spec.Selector)
+	listOptions := metav1.ListOptions{LabelSelector: labelSelector}
+
+	// 获取与 Deployment 匹配的所有 Pods
+	pods, err := clientset.CoreV1().Pods(namespace).List(context.TODO(), listOptions)
+	if err != nil {
+		return nil, fmt.Errorf("error listing pods: %v", err)
+	}
+
+	// 提取 Pod IP 地址
```
Comment (duanmengkk) [inline pkg/kubenest/tasks/anp.go]:
```suggestion
	
```

File: pkg/kubenest/tasks/anp.go [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+	// 获取 Deployment 的标签选择器
+	deployment, err := clientset.AppsV1().Deployments(namespace).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("error getting deployment: %v", err)
+	}
+
+	labelSelector := metav1.FormatLabelSelector(deployment.Spec.Selector)
+	listOptions := metav1.ListOptions{LabelSelector: labelSelector}
+
+	// 获取与 Deployment 匹配的所有 Pods
```
Comment (duanmengkk) [inline pkg/kubenest/tasks/anp.go]:
```suggestion
	
```

File: pkg/kubenest/tasks/anp.go [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+	vcClient, err := getVcClient(client, name, namespace)
+	if err != nil {
+		return "", nil, fmt.Errorf("error when get vcClient, err: %v", err)
+	}
+	return anpAgentManifeattBytes, vcClient, nil
+}
+
+// getDeploymentPodIPs 获取指定 Deployment 的所有 Pod IP 地址
+func getDeploymentPodIPs(clientset clientset.Interface, namespace, deploymentName string) ([]string, error) {
+	// 获取 Deployment 的标签选择器
```
Comment (duanmengkk) [inline pkg/kubenest/tasks/anp.go]:
```suggestion

```

File: pkg/kubenest/util/cert/certs.go [hunk 6]
```diff
@@ -195,6 +195,7 @@ func apiServerAltNamesMutator(cfg *AltNamesMutatorConfig) (*certutil.AltNames, e
 			"kubernetes",
 			"kubernetes.default",
 			"kubernetes.default.svc",
+			"konnectivity-server.kube-system.svc.cluster.local",
```
Comment (duanmengkk) [inline pkg/kubenest/util/cert/certs.go:198]: Can certificate names contain "-"?

File: pkg/kubenest/util/helper.go [hunk 7]
```diff
@@ -214,3 +221,102 @@ func DeleteObject(dynamicClient dynamic.Interface, namespace string, name string
 	}
 	return nil
 }
+
+// DecodeYAML unmarshals a YAML document or multidoc YAML as unstructured
+// objects, placing each decoded object into a channel.
+// code from https://github.com/kubernetes/client-go/issues/216
+func DecodeYAML(data []byte) (<-chan *unstructured.Unstructured, <-chan error) {
```
Comment (duanmengkk) [inline pkg/kubenest/util/helper.go:269]: todo

File: pkg/kubenest/manifest/controlplane/apiserver/mainfests_deployment.go [hunk 8]
```diff
@@ -128,5 +128,356 @@ spec:
...
+      priorityClassName: system-node-critical
+      volumes:
+      - name: kubeconfig
+        secret:
+          defaultMode: 420
+          secretName: {{ .KubeconfigSecret }}
+      - name: varlogkonnectivityserver
+        hostPath:
+          path: /var/log/konnectivity-server.log
+          type: FileOrCreate
```
Comment (OrangeBao) [inline pkg/kubenest/manifest/controlplane/apiserver/mainfests_deployment.go:328]: change ```/var/log/konnectivity-server.log``` to ```/var/log/{namespace}/{name}/konnectivity-server.log``` when use ```hostpath```?

File: pkg/kubenest/manifest/controlplane/apiserver/mainfests_deployment.go (hunk 1 above)
Comment (village-way) [reply pkg/kubenest/manifest/controlplane/apiserver/mainfests_deployment.go]: it used by 
`        - --egress-selector-config-file=/etc/kubernetes/konnectivity-server-config/{{ .Namespace }}/egress_selector_configuration.yaml
`

File: pkg/kubenest/manifest/controlplane/apiserver/mainfests_deployment.go (hunk 2 above)
Comment (village-way) [reply pkg/kubenest/manifest/controlplane/apiserver/mainfests_deployment.go]: fixed

File: pkg/kubenest/tasks/anp.go (hunk 3 above)
Comment (village-way) [reply pkg/kubenest/tasks/anp.go]: fixed

File: pkg/kubenest/tasks/anp.go (hunk 4 above)
Comment (village-way) [reply pkg/kubenest/tasks/anp.go]: fixed

File: pkg/kubenest/tasks/anp.go (hunk 5 above)
Comment (village-way) [reply pkg/kubenest/tasks/anp.go]: fixed

File: pkg/kubenest/util/cert/certs.go (hunk 6 above)
Comment (village-way) [reply pkg/kubenest/util/cert/certs.go:198]: domain name can contain "-"

File: pkg/kubenest/util/helper.go (hunk 7 above)
Comment (village-way) [reply pkg/kubenest/util/helper.go:269]: anp containers use apply to add to apiserver deployment

File: pkg/kubenest/manifest/controlplane/apiserver/mainfests_deployment.go (hunk 8 above)
Comment (village-way) [reply pkg/kubenest/manifest/controlplane/apiserver/mainfests_deployment.go:328]: accept

File: pkg/kubenest/manifest/controlplane/apiserver/mainfests_deployment.go (hunk 1 above)
Comment (OrangeBao) [reply pkg/kubenest/manifest/controlplane/apiserver/mainfests_deployment.go]: > Is this necessary？

This configuration can be removed

Comment (village-way) [reply pkg/kubenest/manifest/controlplane/apiserver/mainfests_deployment.go]: removed

**Pull request merged**